### ISSUE TITLE
Regarding 'desikanKillianyTourvilleLabeling' 

### ISSUE THREAD

**Issue by sj-choi** [ANTsX/ANTsRNet#41]
I am testing 'desikanKillianyTourvilleLabeling' function implemented in ANTsRNet. It works great with huge memory on my virtual machine (VirtualBox). It takes up around 60 GB out of 64 GB. However, it crashes when I ran on other data. When I restart VirtualMachine, then it works great for one data and it crashes again in the middle of processing the next data. I did a few things to refresh the memory. However, it seems not to refresh the memory completely. 

I wonder if I reduce the memory usage by controlling the number of CPU cores used in the process. I anticipate that it will reduce the processing speed but it won't crash by maxing out the memory working on the next data. To test this, I want to reduce the number of CPU cores being used by 'desikanKillianyTourvilleLabeling' but I cannot find that option in the help file. 

Is there any method I can limit the memory usage by desikanKillianyTourvilleLabeling function?

Other segmentation functions (e.g., deepAtropos or deepFalsh) are working great without any issue of crashing (presumably less intensive memory and/or CPU use). 

 Thank you.

**Comment by ntustison** [comment]
You should be able to find stuff online (e.g., [here](https://www.ncaa.com/news/cross-country-men/article/2021-03-07/2021-di-mens-and-womens-cross-country-championship-selections-announced?utm_campaign=inline-superpage))

**Comment by sj-choi** [comment]
The link appears to be about NCAA cross-country championship.

**Comment by ntustison** [comment]
Yeah, I guess that isn't relevant.  Try [this](https://stackoverflow.com/questions/46421258/limit-number-of-cores-used-in-keras).

**Comment by sj-choi** [comment]
I wonder if there is a similar implementation of "desikanKillianyTourvilleLabeling" for "pure" ANTs (not using ANTsRNet/TensorFlow under the hood). 

What I am looking for is ANTs command/function that can do DKT labeling in a BASH script using ANTs functions. (neither R nor Python) I checked "${ANTSPATH}" but I couldn't see nor recognize it. If it was already there, could you let me know? It might have a different name, I guess.

**Comment by ntustison** [comment]
No.  You could use joint label fusion (``antsJointFusion``) to get a set of DKT labels but it's not a similar implementation.

**Comment by sj-choi** [comment]
Thanks for your clarification.
By the way, reducing the number of cores was not the solution. A critical issue appears to be the physical memory size being used during the DKT labeling using ANTsRNet. 

- I wonder if there is any recommended memory size for this process?

- Is the DKT labeling only available for ANTsRNet/ANTsPyNet? Is a DKT implementation available in ANTsR, not ANTsRNet? If so, I hope it uses less memory.

**Comment by ntustison** [comment]
>  I wonder if there is any recommended memory size for this process?

I would recommend you do some searching to handle any system-specific memory issues.

>  Is the DKT labeling only available for ANTsRNet/ANTsPyNet? Is a DKT implementation available in ANTsR, not ANTsRNet? If so, I hope it uses less memory.

No, the DKT implementation is deep learning-based.  As such, it requires the keras backend only accessible through ANTsRNet/ANTsPyNet.

**Comment by JinFeng-Wu** [comment]
I ran desikanKillianyTourvilleLabeling for sevaral times with the default options (doLobar = TRUE), but this functionality produced around 8 ROI stats, can you help to check it out?
